### PR TITLE
Pre commit

### LINKED ISSUE #13: Set up github actions
@klindsay28 ran `black` on the branch merged in #12 -- we should setup github actions to continue those checks. Also useful:

1. A README file outlining how to set up pre-commit hooks to also run black
1. Including `pytest` in the environment and running some tests as part of the github action (some of the code brought in with #12 came from another repository that already [has tests](https://github.com/klindsay28/CESM2_coup_carb_cycle_JAMES/blob/master/tests/test_utils_units.py) we can copy)

### PR DESCRIPTION
**Pull request by mnlevy1981**
Adding `pre-commit` to the environment and a `README.md` file detailing how to set it up. I want to add `pytest` via this pull request as well (and have pre-commit run the tests as well), so this will remain a draft PR until I finish that.

Fixes #13

### CODE REVIEW

**Comment by mnlevy1981** [comment]
Note that this will downgrade python from 3.8 to 3.7; it looks like there's an issue with `virtualenv` and py3.8, and even with workarounds like downgrading `virtualenv` from v20 to v16 I couldn't get things to work.

**Comment by mnlevy1981** [comment]
I think this one is ready to go, but would appreciate more eyes on it. There's a README with instructions for setting up `pre-commit`, and then github actions for running `black` and `pytest`. The `pytest` tests are just a starting point, we should probably write more as we continue to develop the code, but I think it's sufficient for this initial PR

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        hooks:
+            -   id: trailing-whitespace
+            -   id: end-of-file-fixer
+            -   id: check-yaml
+
+    -   repo: https://github.com/ambv/black
+        rev: 19.10b0
+        hooks:
+            -   id: black
+                args: []
```
Comment (andersy005) [inline .pre-commit-config.yaml:13]: Could we add the following pre-commit hook for formatting notebooks as well:

```yaml
  - repo: https://github.com/deathbeds/prenotebook
    rev: f5bdb72a400f1a56fe88109936c83aa12cc349fa
    hooks:
      - id: prenotebook
        args:
          [
            '--keep-output',
            '--keep-metadata',
            '--keep-execution-count',
            '--keep-empty',
          ]
```

**Comment by mnlevy1981** [comment]
As noted in the last commit message, I've implemented two of Anderson's suggestions:

1. Added [a prenotebook hook](https://github.com/marbl-ecosys/HiRes-CESM-analysis/pull/17#discussion_r478643213)
1. Gihub Action [now relies on pre-commit](https://github.com/marbl-ecosys/HiRes-CESM-analysis/pull/22#issuecomment-682146726)

`prenotebook` did catch a few issues which are fixed on this branch and will likely cause conflicts when merged into other branches... but I think the right thing to do is keep the existing version on the branch and then let `pre-commit` fix it.

I also updated the environment file to get `precommit` from `pip` instead of `conda` -- I think this is necessary to have some consistency between `pre-commit` and our `(hires-marbl)` conda environment. The README file has a short section on how to make sure `pre-commit` is using the latest version; as far as I can tell, there needs to be _some_ manual intervention to update the version in `pre-commit` when the version in the environment changes but I don't think that will turn out to be an issue in practice.

**Pull request merged**